addon/model-data.js

```diff
@@ -93,25 +84,19 @@ export default class M3ModelData {
 
     // Properties related to child model datas
     this._parentModelData = parentModelData;
-    this._embeddedInternalModel = embeddedInternalModel;
+    this._embeddedInternalModel = null;
     this.__childModelDatas = null;
     this._schema = SchemaManager;
 
     this.schemaInterface = new M3SchemaInterface(this);
 
     // Properties related to projections
     this._baseModelName = this._schema.computeBaseModelName(this.modelName);
-    this._baseModelData = null;
+    this._baseModelData = baseModelData;
     this._projections = null;
 
-    if (parentKey !== undefined && parentKey !== null) {
-      this._parentModelData._addChildModelData(parentKey, parentValueIsArray, this);
-    }
-
-    if (this._baseModelName) {
+    if (this._baseModelName || this._baseModelData) {
```

Comment:
This can happen, when a nested property has been projected as its base model, e.g. the `address` field of a company is projected as `Address` instead of `CountryOnlyAddress`.